tests/test_serialization.py

```diff
@@ -0,0 +1,153 @@
+import json
+from typing import Type
+
+import attrs
+import pytest
+
+from web_poet import HttpResponse, HttpResponseBody, ResponseUrl, WebPage
+from web_poet.page_inputs.url import _Url
+from web_poet.serialization import (
+    SerializedLeafData,
+    deserialize,
+    deserialize_leaf,
+    read_serialized_data,
+    register_serialization,
+    serialize,
+    serialize_leaf,
+    write_serialized_data,
+)
+
+
+def _assert_webpages_equal(p1: WebPage, p2: WebPage) -> None:
+    assert p1.response.body == p2.response.body
+    assert p1.response.status == p2.response.status
+    assert p1.response.headers == p2.response.headers
```

Comment:
Similar to when asserting if the URL types are the same, I think we could also compare if they are of the same type (e.g. `HttpResponseBody` == `HttpResponseBody`.)